template/utils/misc.py

```diff
@@ -1,39 +1,41 @@
 # {% include 'template/license_header' %}
 
 import numpy as np
-from sklearn.metrics import accuracy_score, recall_score, precision_score, f1_score
-
+from datasets import load_metric
 from zenml.enums import StrEnum
 
-def compute_metrics(p):
-    pred, labels = p
-    pred = np.argmax(pred, axis=1)
-
-    accuracy = accuracy_score(y_true=labels, y_pred=pred)
-    recall = recall_score(y_true=labels, y_pred=pred, average='micro')
-    precision = precision_score(y_true=labels, y_pred=pred, average='micro')
-    f1 = f1_score(y_true=labels, y_pred=pred, average='micro')
 
-    return {"accuracy": accuracy, "precision": precision, "recall": recall, "f1": f1}
+def compute_metrics(eval_pred: tuple) -> dict[str, float]:
+    """Compute the metrics for the model.
 
+    Args:
+        eval_pred: The evaluation prediction.
 
+    Returns:
+        The metrics for the model.
+    """
+    logits, labels = eval_pred
+    predictions = np.argmax(logits, axis=-1)
+    # calculate the mertic using the predicted and true value
+    accuracy = load_metric("accuracy").compute(
+        predictions=predictions, references=labels
+    )
+    f1 = load_metric("f1").compute(
+        predictions=predictions, references=labels, average="weighted"
+    )
+    precision = load_metric("precision").compute(
+        predictions=predictions, references=labels, average="weighted"
+    )
+    return {"accuracy": accuracy, "f1": f1, "precision": precision}
 
-def find_max_length(dataset):
-    return len(max(dataset, key=lambda x: len(x.split())).split())
 
+def find_max_length(dataset: list[str]) -> int:
+    """Find the maximum length of the dataset.
```

Comment:
Maybe this needs a bit more context in the docstring?

HF dataset? (since I guess it's unusual for datasets to be in a list of strings...)